### DISCUSSION TITLE
Bootstrapper applications uninstall shortcut

### DISCUSSION THREAD

**Discussion by ad-chuliang** [wixtoolset/issues#8664] [Questions]
I know how to create msi uninstall_shortcut from "https://wixtoolset.org/docs/v3/howtos/files_and_registry/create_uninstall_shortcut/".
However, how to create a bootstrapper applications uninstall shortcut?

**Comment by chancezheng** [comment]
https://github.com/CookCSharp/CookPopularInstaller/blob/master/CookPopularInstaller.Msi/Product.wxs, move to bottom

**Reply by ad-chuliang** [reply]
Are you referring to lines 206 to 212? That use msiexec to wake msi matain dialog, not for my question. When I use Bundle to include a msi, the shortcut of bootstrapper application exe is what I want.

**Reply by chancezheng** [reply]
you can write a uninstall app to wrap, such as console app,

**Reply by ad-chuliang** [reply]
How to link to Bundle?

**Reply by chancezheng** [reply]
you can find the package cache path by registry, usually it is locate at "%programdata%\Package Cache", then you can use your wrap app to start it, that is bundle exe

**Comment by bevanweiss** [comment]
Bootstrappers and bundles aren't intended to deploy their own content, they just launch other things to deploy the actual files.
So you could create a very minimal MSI that just installs the shortcut.  The shortcut will just point back to the bundle and have a command line of --uninstall or similar.
The location of the cache for the bundle will generally be in ProgramData somewhere, you might need to have the bundle pass into the MSI the location that it is stored, in case someone overrides the cache location.

**Reply by ad-chuliang** [reply]
Thanks, I get what your mean. But, how to have the bundle pass into the MSI the location that it is stored? I found that Bundle has no child element like ProductCode which is in MSI. The value of UninstallString key which in registery is like : "...path.../setup.exe" /uninstall. And, maybe I also can not change the value of UninstallString key which belong to MSI in the path of "HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\" , because I have tried to change that value but failed and maybe I used wrong way. So, could you give me the detail operation.

**Reply by wmanning** [reply]
I think you use a child MsiProperty element.
https://wixtoolset.org/docs/v3/xsd/wix/msipackage/
https://wixtoolset.org/docs/v3/xsd/wix/msiproperty/

So create a MsiPackage to install your MSI and then under that a MsiProperty to set the directory.  Then use that property in the MSI.

**Reply by wmanning** [reply]
above links are for v3 wix.

v4/5 docs:
https://wixtoolset.org/docs/schema/wxs/msppackage/
https://wixtoolset.org/docs/schema/wxs/msiproperty/
